### ISSUE TITLE
clink.1.2.17.b5751c_setup.exe is blocked by windows defender

### ISSUE THREAD

**Issue by bw1faeh0** [chrisant996/clink#138]
Windows defender blocks the clink.1.2.17.b5751c_setup.exe because of a detected thread:

Trojan:Win32/Emali.A!cl
file: C:\Users\chris\Downloads\clink.1.2.17.b5751c_setup.exe

**Comment by chrisant996** [comment]
There's nothing Clink can do about that. You can report it to Windows Defender or set up an exclusion for it.

FWIW, for me Windows Defender is not detecting any malware and runs the setup.exe without errors or complaints.

**Comment by chrisant996** [comment]
Oh, all of a sudden now Windows Defender is complaining.  It did not complain initially when 1.2.17.b5751c was released.

So it's a bug in the Windows Defender signatures.  There's nothing Clink can do about that.  We just have to report it to Windows Defender and rely on they to fix their bug.

**Comment by chrisant996** [comment]
(This has prompted me to resume exploring the options for signing the executable files.  It seems that an EV code signing certificate might help avoid SmartScreen false positives, but it's impossible for me to obtain an EV code signing certificate because I am not a company.  It seems that neither an EV cert nor a normal cert are effective for avoiding false positives with anti-malware suites.  Nevertheless, if I can find a way to set up code signing for less than a couple hundred dollars per year, I might end up signing the binaries anyway.)

**Comment by bw1faeh0** [comment]
Ok, I just wanted to double check. I'll try the zip file next time.

**Comment by chrisant996** [comment]
Windows Defender has permanently deleted all Clink installers off my computer, including old archived ones.
I am awaiting a response from Windows Defender about the false positive.

It is also ignoring the exclusion I set up for my `c:\repos\clink` folder.
This has made it impossible for me to produce setup.exe installers anymore.

**Comment by chrisant996** [comment]
Windows Defender responded, and it should no longer be blocking the Clink installer program.  (Of course a false positive could recur in the future, though.)

**Comment by nathaniel-holder** [comment]
Windows Defender still blocking:
![image](https://user-images.githubusercontent.com/3880337/141682194-1ea10db4-0726-4d08-9c2e-85f739639671.png)

Here's Windows Defender version info:
![image](https://user-images.githubusercontent.com/3880337/141682205-3a08bb89-6748-454e-9541-e71d55a7cbf4.png)

**Comment by chrisant996** [comment]
You have to report the issue to Windows Defender (I.e. to the publisher of the anti virus suite that is incorrectly flagging Clink).

There is nothing Clink can do about this.  (If there were, then every virus would do it to make themselves seem good.)

**Comment by nathaniel-holder** [comment]
Isn't the issue that there is no publisher?

**Comment by chrisant996** [comment]
> Isn't the issue that there is no publisher?

Oh this isn't anti virus at all. This is not a duplicate of (or even related to) this issue 138.

See explanation in #198.

Fell free to read up on SmartScreen and code signing to learn more.
